contracts/MechMarketplace.sol

```diff
@@ -446,6 +435,54 @@ contract MechMarketplace {
         ));
     }
 
+    /// @dev Checks for staking instance contract validity.
+    /// @param stakingInstance Staking instance address.
+    /// @param serviceId Service Id.
+    function checkStakingInstance(address stakingInstance, uint256 serviceId) public view {
```

Comment:
This function verifies that the staking contract of either requester / mech is valid and that the corresponding service is staked.